components/content_settings/core/browser/ad_block_only_mode_content_settings_utils_unittest.cc

```diff
@@ -0,0 +1,209 @@
+                                                 GURL("https://example.com"),
+                                                 ContentSettingsType::COOKIES);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_COOKIES);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_REFERRERS);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_ADS);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_BLOCK);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_TRACKERS);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_COSMETIC_FILTERING);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_FINGERPRINTING_V2);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
+  }
+
+  {
+    base::AutoLock auto_lock(ad_block_only_mode_rules.GetLock());
+    auto rule = ad_block_only_mode_rules.GetRule(
+        GURL("https://example.com"), GURL("https://example.com"),
+        ContentSettingsType::BRAVE_REMEMBER_1P_STORAGE);
+    ASSERT_TRUE(rule);
+    EXPECT_EQ(ValueToContentSetting(rule->value), CONTENT_SETTING_ALLOW);
+    CheckMetadata(rule->metadata);
```

Comment:
These tests appear to duplicate, maybe a test util such as VerifyRuleExpectation?

Reply:
Changed